src/controller.js

```diff
@@ -92,58 +93,68 @@ export function getController (config) {
   /*
    * Prepare scenes data.
    */
-  _config.scenes = config.scenes.map((scene, index) => {
-    scene.index = index;
-
-    if (scene.viewSource && (typeof scene.duration === 'string' || scene.start?.name)) {
-      scene = getTransformedScene(scene, root, viewportSize, horizontal, absoluteOffsetContext);
-
+  _config.scenes = Object.values(
+    config.scenes.reduce((acc, scene, index) => {
```

Comment:
Let's get wild and try using [`Object.groupBy()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy)?

Reply:
waiting for now, added a TODO for finding a polyfill and using groupBy